mne_icalabel/gui/_label_components.py

```diff
@@ -0,0 +1,236 @@
+        self.connect_signals_to_slots()
+
+    @staticmethod
+    def list_components(ica):
+        """List the components in a QListView."""
+        list_components = QListWidget()
+        list_components.setSelectionMode(QAbstractItemView.SingleSelection)
+        list_components.setMinimumWidth(6 * _CH_MENU_WIDTH)
+        list_components.setMaximumWidth(6 * _CH_MENU_WIDTH)
+        list_components.addItems([f"ICA{str(k).zfill(3)}" for k in range(ica.n_components_)])
+        return list_components
+
+    def connect_signals_to_slots(self):  # noqa: D102
+        self.list_components.clicked.connect(self.list_component_clicked)
+        self.buttonGroup_labels.buttonGroup.buttons()[-1].clicked.connect(self.reset)
+
+    @Slot()
+    def list_component_clicked(self):
+        """Jump to the selected component and draw the plots."""
+        self.update_saved_labels()
+        self._reset_all_buttons()
+
+        # reset figures
+        self.widget_topo.reset()
+        self.widget_psd.reset()
+
+        # update selected IC
+        self._current_ic = self.list_components.currentRow()
+
+        # create dummy axes
+        dummy_fig, dummy_axes = plt.subplots(3)
+        axes = [
+            self.widget_topo.axes,
+            dummy_axes[0],
+            dummy_axes[1],
+            self.widget_psd.axes,
+            dummy_axes[2],
+        ]
+        self._ica.plot_properties(self._inst, axes=axes, picks=self._current_ic, show=False)
+        del dummy_fig
+
+        # update figures
+        self.widget_topo.redraw()
+        self.widget_psd.redraw()
+
+        # update selected label if one was saved
+        if self._current_ic in self.saved_labels:
+            label = self.saved_labels[self._current_ic]
+            idx = self.labels.index(label)
+            self.buttonGroup_labels.buttonGroup.button(idx).setChecked(True)
+
+    def update_saved_labels(self):
+        """Update the labels saved."""
+        selected = self.buttonGroup_labels.buttonGroup.checkedButton()
+        if selected is not None:
+            self.saved_labels[self._current_ic] = selected.text()
+
+    @Slot()
+    def reset(self):
+        """Action for the reset button."""
+        self._reset_all_buttons()
+        if self._current_ic in self.saved_labels:
+            del self.saved_labels[self._current_ic]
+
+    def _reset_all_buttons(self):
+        """Reset all buttons."""
+        self.buttonGroup_labels.buttonGroup.setExclusive(False)
+        for button in self.buttonGroup_labels.buttonGroup.buttons():
+            button.setEnabled(True)
+            button.setChecked(False)
+        self.buttonGroup_labels.buttonGroup.setExclusive(True)
+
+    def closeEvent(self, event):
+        """Clean up upon closing the window.
+
+        Check if any IC is not labelled and ask the user to confirm if this is
+        the case. Save all labels in BIDS format.
+        """
+        self.update_saved_labels()
+        print (self.saved_labels)
```

Comment:
@adam2392 Here it is, instead of a print we need to export the saved labels. This event corresponds to "closing the GUI".

Reply:
Hi  @mscheltienne. Adding this now. 

So one issue is that we need the filename to export to BIDS. There are two ways I could see a user doing this:

1. If they load in the raw/ica and it's already in BIDS, instead of passing in the instance, we pass in the BIDS filenames. Then we can extract the BIDS path and save upon closing without requiring the user to pass in a filename because it would just be auto-saved to the "location" it needs to be saved. 
2. If they load in the raw/ica instance, then we need the corresponding filename.

1 is kind of constraining because it requires the users to store the data in exactly the way we need/expect. 2 requires adding an additional UX pipeline for the user to add either the BIDS entities to infer the final filename, or actually add the output filename.

WDYT?

Reply:
Actually, a 3rd option is:

3. If they load in the raw/ica instance, we just modify the ica instance to have labels in the backend. I think, we can populate the `labels_` dictionary of the ICA instance in place: https://mne.tools/stable/generated/mne.preprocessing.ICA.html. Once users close the labeling GUI, then they can run `write_components_tsv` themselves. Or we can add a save-as button.

^ basically, saving automatically to disc when closing the file isn't ideal.

This might require https://github.com/mne-tools/mne-python/issues/9846.

Reply:
I'll think about it, for now, I was more inclined towards (1) because it was constrained. Supporting only datasets in BIDS format makes this first version simpler. 

I don't think it's a good idea to have a pop-up GUI to ask the user for a BIDS path, it can be more error-prone and I don't see a reason to save the IC labels in a .tsv sidecar in a BIDS folder if the original data and ICA decomposition are not in the same BIDS folder.

I like option (3) as well, modifying the ICA instance in place makes a lot of sense, and does not entirely make the .tsv redundant as it will also contain the information about who authored the labels (and other information).

Reply:
Yeah the more I think about it...

I'm inclined towards 3) with the pipeline modifying ICA in place, and then expecting the user to save it themselves to BIDS using the API provided in mne-icalabel. The MNE-Python API should eventually support saving the ICA instance with labels, so hypothetically someone might not work in BIDS. 

Optionally, we can also provide a menu inside the GUI to "save-as" and the user can specify the file. 

I think to move this forward, I'll just hack the ICA instance and then provide an example showing how we can save it using the above pipeline. Eventually, we would like to make this pipeline possible via the PR into MNE-Python and FIFF-constants.

Reply:
Agree, that makes the most sense.